### ISSUE TITLE
caption generation

### ISSUE THREAD

**Issue by lihao612** [NJUVISION/ResULIC#12]
For the image 0057129.png in the LSDIR dataset, the author's published title is: "A group of people are walking through a cave, with one person standing in the entrance and others following behind. The cave has a rocky texture and is dimly lit, creating a mysterious atmosphere." 
However, the result generated by qwen.py is: "The image depicts a natural cave with a rugged, textured interior. The cave's ceiling and walls are uneven, showing the raw, earthy tones of the rock. A large opening in the cave reveals a bright, outdoor landscape with greenery and distant hills under a clear sky. Inside the cave, three individuals are visible near the opening. The person on the left wears a striped shirt." 
The results are clearly different, and even when the prompts are changed to those in the paper's description, the generated results remain inconsistent. Could the author please release the correct title generation code?

**Comment by KangLEkk** [comment]
Thank you for your feedback. This dataset was created relatively early, and at the time, the prompts were extracted using Llava1.5, which may lead to inconsistencies compared to the results obtained with Qwen. Unfortunately, we experienced a data loss incident during the process, and the original generation code could not be recovered. I recommend that you use a more advanced VLM to refine this batch of prompts or consider using a more recent dataset, as this may yield better results.

**Comment by lihao612** [comment]
Thank you for your reply. I am currently reproducing your paper. Could you please tell me if gpt4o is used to generate the text during the testing phase? Could you provide the text generation code that conforms to the paper's specifications?

Also, are the experimental results obtained during the testing phase using text generated based on gpt4o?Is the clic2020 dataset used in the test the clic2020_professional_test version?

Could you please give the parameter settings for ddim_steps, ddim_eta, Q, and add_steps for the author's publicly available results indicator?

**Comment by Linkon87** [comment]
> Thank you for your reply. I am currently reproducing your paper. Could you please tell me if gpt4o is used to generate the text during the testing phase? Could you provide the text generation code that conforms to the paper's specifications?
> 
> Also, are the experimental results obtained during the testing phase using text generated based on gpt4o?Is the clic2020 dataset used in the test the clic2020_professional_test version?
> 
> Could you please give the parameter settings for ddim_steps, ddim_eta, Q, and add_steps for the author's publicly available results indicator?

I have the same question. Could you please provide an example of the test parameters? @KangLEkk

**Comment by Linkon87** [comment]
> > Thank you for your reply. I am currently reproducing your paper. Could you please tell me if gpt4o is used to generate the text during the testing phase? Could you provide the text generation code that conforms to the paper's specifications?
> > Also, are the experimental results obtained during the testing phase using text generated based on gpt4o?Is the clic2020 dataset used in the test the clic2020_professional_test version?
> > Could you please give the parameter settings for ddim_steps, ddim_eta, Q, and add_steps for the author's publicly available results indicator?
> 
> I have the same question. Could you please provide an example of the test parameters? [@KangLEkk](https://github.com/KangLEkk)

Additionally, how do I load the model trained in the first stage to train the second stage? Are there any relevant examples?Thank you very much for your outstanding work. If you could provide an example of how to reproduce it, I believe this work would have an even greater impact.🎉 @KangLEkk

**Comment by KangLEkk** [comment]
> Thank you for your reply. I am currently reproducing your paper. Could you please tell me if gpt4o is used to generate the text during the testing phase? Could you provide the text generation code that conforms to the paper's specifications?
> 
> Also, are the experimental results obtained during the testing phase using text generated based on gpt4o?Is the clic2020 dataset used in the test the clic2020_professional_test version?
> 
> Could you please give the parameter settings for ddim_steps, ddim_eta, Q, and add_steps for the author's publicly available results indicator?

Thank you for your interest in our work, and I apologize for not responding sooner.

1. In the testing phase of our paper, we used GPT-4o. The usage method is almost identical to that in qwen.py, as both involve API calls. Their performance is nearly identical, with Qwen even having a slight advantage. Currently, we continue to use the setup in qwen.py for subsequent research. Therefore, I recommend you follow the qwen.py script.

2. The CLIC2020 dataset used is the clic2020_professional_test version, which is a test set containing 428 images.

3. Regarding the parameter settings for testing:

-   add_steps should be set according to the corresponding configuration file.
-   ddim_steps is generally set to 3 by default. (Note: add_steps must be divisible by ddim_steps.)
-   ddim_eta is uniformly set to 0, corresponding to deterministic sampling, as mentioned in our paper.
-   Q is a parameter set for file saving purposes and has no practical effect.

- For example, when testing with the provided 0.002 bpp model (corresponding config file: configs/model/stage2/1_1_24/cldm_eps_600_ddim.yaml), you can set the parameters as follows: add_steps = config.params.add_steps (which is 600 here), ddim_steps = 3, ddim_eta = 0

4. For the second stage of training, use the provided train_stage2.py script directly. Regarding loading the pre-trained model, taking the 0.002 bpp model as an example, you need to modify the configs/train_stage2.yaml file as follows during the second stage of training:
```
model:
  config: configs/model/stage2/1_1_24/cldm_eps_600_ddim.yaml
  resume: ~ (Please enter the path to your pre-trained model here)
```

**Comment by lihao612** [comment]
Hello author, during my testing of the public model, I found that the bitrate was about 0.015 higher than your result. Is this because the generated text is too long (around 40 words)? I didn't modify the image pre-compression process at all.

And the dists index is about 0.018 higher than the publicly available results. Is this normal?

**Comment by KangLEkk** [comment]
> Hello author, during my testing of the public model, I found that the bitrate was about 0.015 higher than your result. Is this because the generated text is too long (around 40 words)? I didn't modify the image pre-compression process at all.
> 
> And the dists index is about 0.018 higher than the publicly available results. Is this normal?

Sorry, I didn’t quite understand what you meant. If you tested using the 0.01bpp model we provided, the 0.015bpp you mentioned is a normal value.

Regarding the text length, it might be because the extracted original caption was very long, so the text remained somewhat lengthy even after residual processing. 

As for the dists being 0.018 higher, this is normal and falls within the range of acceptable error. Differences in GPU precision or variations in prompts may all affect the results.

**Comment by lihao612** [comment]
I tested a 0.03bpp model.Is this bpp the total bit rate of the residual version?In this case, bpp would be almost the same.
And could you please provide a download link for clic2020_professional_test version?

**Comment by KangLEkk** [comment]
> I tested a 0.03bpp model.Is this bpp the total bit rate of the residual version?In this case, bpp would be almost the same.
> And could you please provide a download link for clic2020_professional_test version?

Yes, the total bitrate is calculated based on the residual version, so it is normal for there to be some discrepancy compared to your test results.

Additionally, the CLIC2020 test set can be found at the following links:
https://data.vision.ee.ethz.ch/cvl/clic/test/CLIC2020Professional_test.zip
https://data.vision.ee.ethz.ch/cvl/clic/test/CLIC2020Mobile_test.zip

One thing to note: I'm not entirely sure whether the commonly used 428 images are from the CLIC2020Professional_test only or from the combined set of CLIC2020Professional_test and CLIC2020Mobile_test. You may need to verify this.

**Comment by lihao612** [comment]
Thank you. If that's the case, the bpp results can be considered the same. However, my PSNR is 0.014 lower and my LPIPS is 0.008 higher. Is there something wrong?

**Comment by Linkon87** [comment]
> > I tested a 0.03bpp model.Is this bpp the total bit rate of the residual version?In this case, bpp would be almost the same.
> > And could you please provide a download link for clic2020_professional_test version?
> 
> Yes, the total bitrate is calculated based on the residual version, so it is normal for there to be some discrepancy compared to your test results.
> 
> Additionally, the CLIC2020 test set can be found at the following links: https://data.vision.ee.ethz.ch/cvl/clic/test/CLIC2020Professional_test.zip https://data.vision.ee.ethz.ch/cvl/clic/test/CLIC2020Mobile_test.zip
> 
> One thing to note: I'm not entirely sure whether the commonly used 428 images are from the CLIC2020Professional_test only or from the combined set of CLIC2020Professional_test and CLIC2020Mobile_test. You may need to verify this.

Is the COCO dataset used for testing coco-30-val-2014? This dataset seems to have 30,000 images, and a corresponding caption is generated for each image?

**Comment by Linkon87** [comment]
Thank you. What size model is used for image descriptions with Qwen? I can't seem to find an open-source model for Qwen-VL-MAX; is it necessary to use the official token? Is it possible to use a locally deployed model to obtain image descriptions, such as Qwen3-VL-8B-Instruct? Would this have a performance impact? @KangLEkk

**Comment by KangLEkk** [comment]
> Thank you. If that's the case, the bpp results can be considered the same. However, my PSNR is 0.014 lower and my LPIPS is 0.008 higher. Is there something wrong?

This is entirely normal, and such variations can be considered within the acceptable margin of error. Different GPUs or different random seeds may all lead to varying results.

**Comment by KangLEkk** [comment]
> Thank you. What size model is used for image descriptions with Qwen? I can't seem to find an open-source model for Qwen-VL-MAX; is it necessary to use the official token? Is it possible to use a locally deployed model to obtain image descriptions, such as Qwen3-VL-8B-Instruct? Would this have a performance impact? [@KangLEkk](https://github.com/KangLEkk)

The primary testing in our paper was conducted via API calls, though locally deployed models such as Qwen3-VL-8B-Instruct can also be used. In our experiments, we also employed a locally deployed Llama-3.2-11B model, which still fulfilled our requirements, albeit with potentially reduced performance compared to larger models.
